### ISSUE TITLE
runtime: TestNeedmDeadlock failure with SIGSEGV on linux-mips-rtrk builder

### ISSUE THREAD

**Issue by bcmills** [golang/go#43625]
[2021-01-08T18:15:43-8f6a9ac/linux-mips-rtrk](https://build.golang.org/log/b84d0579bf5fcd426c4d450bd6749f33c5077d9f)

```
--- FAIL: TestNeedmDeadlock (0.03s)
    crash_test.go:104: /tmp/gobuilder-mips64/tmp/go-build2938167109/testprogcgo.exe NeedmDeadlock exit status: exit status 2
    crash_cgo_test.go:631: want OK
        , got fatal error: unexpected signal during runtime execution
        [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x0]
        
        runtime stack:
        runtime.throw(0x57cb56, 0x2a)
        	/tmp/gobuilder-mips64/go/src/runtime/panic.go:1112 +0x60
        runtime.sigpanic()
        	/tmp/gobuilder-mips64/go/src/runtime/signal_unix.go:714 +0x3b4
        
        goroutine 17 [syscall, locked to thread]:
        runtime.goexit()
        	/tmp/gobuilder-mips64/go/src/runtime/asm_mipsx.s:635 +0x4 fp=0x83aff0 sp=0x83aff0 pc=0x47d9f8
        
        goroutine 1 [syscall]:
        main._Cfunc_runNeedmSignalThread()
        	_cgo_gotypes.go:404 +0x4c
        main.NeedmDeadlock()
        	/tmp/gobuilder-mips64/go/src/runtime/testdata/testprogcgo/needmdeadlock.go:93 +0x4c
        main.main()
        	/tmp/gobuilder-mips64/go/src/runtime/testdata/testprogcgo/main.go:34 +0x1d8
        
        goroutine 6 [runnable]:
        time.Sleep(0x1, 0x2a05f200)
        	/tmp/gobuilder-mips64/go/src/runtime/time.go:184 +0x184
        main.NeedmDeadlock.func1()
        	/tmp/gobuilder-mips64/go/src/runtime/testdata/testprogcgo/needmdeadlock.go:88 +0x48
        created by main.NeedmDeadlock
        	/tmp/gobuilder-mips64/go/src/runtime/testdata/testprogcgo/needmdeadlock.go:87 +0x44
        
FAIL
FAIL	runtime	116.006s
```

CC @ianlancetaylor @mengzhuo

**Comment by bcmills** [comment]
This test is new as of Go 1.16 (for #42207), so marking as release-blocker at least until we better understand the failure mode here. (CC @golang/release)

**Comment by mengzhuo** [comment]
Thanks @bcmills 

I'll take a look.

**Comment by aclements** [comment]
(FYI, we dropped the release-blocker label because this is a non-first-class port.)

**Comment by mengzhuo** [comment]
Some updates:
Unfortunately I can't reproduce this issue on my loongson box(mips64le) using 32bit compatible mode.

**Comment by dmitshur** [comment]
Thanks for the update. It seems there's not enough time to become able to reproduce and then fix this for Go 1.16, so I'll reset the milestone to Backlog for now.
